### PR TITLE
Convert annotations provided by the new parser to IR

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Builds on top of #3780 and converts the new `Tree.Annotated` to IR.
[ci no changelog needed]

### Checklist

Please include the following checklist in your PR:

- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

**Review by kazcw — APPROVED** [review]
Looks good. Does it handle this case?

```ruby
normalize self = @Builtin_Method "File.normalize"
```

**Comment by JaroslavTulach** [comment]
Looks like it handles:
> ```ruby
> normalize x = @Builtin_Method "File.normalize"
> ```
but it doesn't handle `self`! That's something I solve in other PR.

File: engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java [hunk 1]
```diff
@@ -947,6 +958,30 @@ yield switch (ast) {
...
+        var fnAsArg = translateCallArgument(fn.getArg(), insideTypeSignature);
+        yield translateAnnotation(ir, fn.getFunc(), cons(fnAsArg, callArgs));
+      }
+      case Tree.ArgumentBlockApplication fn -> {
+        var fnAsArg = translateCallArgument(fn.getLhs(), insideTypeSignature);
+        var arg = translateCallArgument(expr, insideTypeSignature);
+        callArgs = cons(fnAsArg, cons(arg, callArgs));
+        yield translateAnnotation(ir, null, callArgs);
+      }
+      case null -> {
```
Comment (hubertp) [inline engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java:974]: That is a bit surprising case

**Review by hubertp — APPROVED** [review]

File: engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java:974]: Line 972 and line 980 reach it via `yield translateAnnotation(ir, null, callArgs);`

**Pull request merged**